unistore-server/main.go

```diff
@@ -248,14 +248,17 @@ func createDB(subPath string, safePoint *tikv.SafePoint, conf *config.Engine) *b
 	opts.ValueLogWriteOptions.WriteBufferSize = 4 * 1024 * 1024
 	opts.Dir = filepath.Join(conf.DBPath, subPath)
 	opts.ValueDir = opts.Dir
-	opts.TableLoadingMode = options.MemoryMap
 	opts.ValueLogFileSize = conf.VlogFileSize
-	opts.ValueLogLoadingMode = options.FileIO
 	opts.MaxTableSize = conf.MaxTableSize
 	opts.NumMemtables = conf.NumMemTables
 	opts.NumLevelZeroTables = conf.NumL0Tables
 	opts.NumLevelZeroTablesStall = conf.NumL0TablesStall
 	opts.SyncWrites = conf.SyncWrite
+	for i := range opts.TableBuilderOptions.CompressionPerLevel {
+		opts.TableBuilderOptions.CompressionPerLevel[i] = config.ParseCompression(conf.Compression[i])
```

Comment:
Set `CompressionPerLevel` with `make([]options.CompressionType, len(conf.Compression[i])`? Otherwise we changed the value in `badger.DefaultOptions`.

Reply:
No, `badger.DefaultOptions` is a struct value, its copied to `opts`.

Reply:
`CompressionPerLevel` is `[]options.CompressionType`, opts only copied the fat pointer of the slice but shared the same underlying memory.